### ISSUE TITLE
Search fails if term is a tag and comes after a numeric tag

### ISSUE THREAD

**Issue by myrho** [squidfunk/mkdocs-material#7750]
### Bug description

Search fails if the term is a tag and comes after a numeric tag in a documents tags list. 

In the console I get:
```
Invalid query: k – see https://bit.ly/2s3ChXG
TypeError: s.slice is not a function
```

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
- https://github.com/squidfunk/mkdocs-material/pull/4281

### Reproduction

[9.5.46-search-fails-if-term-appears-as-a-tag-after-a-numeric-tag.zip](https://github.com/user-attachments/files/17905977/9.5.46-search-fails-if-term-appears-as-a-tag-after-a-numeric-tag.zip)

### Steps to reproduce

Install as in the reproduction zip.

Perform a search for "foo". No results are found although the index.html should be listed in the results.

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) using the [built-in info plugin](https://squidfunk.github.io/mkdocs-material/plugins/info/).

**Comment by squidfunk** [comment]
Thanks for reporting! Fixed in d4683447. I'm not sure why this did not occur until now, because it is such a common case. The problem was that numeric tags where not converted to strings, which does not work with a search that works on strings.

**Comment by squidfunk** [comment]
Released as part of [9.5.47](https://squidfunk.github.io/mkdocs-material/changelog/#9.5.47).
